### ISSUE TITLE
Persistent no reachable server while connecting to a legitimate replica set

### ISSUE THREAD

**Issue by logicaleak** [go-mgo/mgo#371]
Hello,

I have a replica set from mongodb atlas, to which I can connect with ANY other language, and regular mongo client, with the url provided with the format : 

"mongodb://user:pass@prefix1.mongodb.net:27017,prefix2.mongodb.net:27017,prefix3.mongodb.net:27017/test?&replicaSet=Cluster0-shard-0&authSource=admin"

No matter what I tried, adding ssl=true and removing, nothing works. It is always "no reachable server". 

I tried every single combination for url, every combination for dialConfig, and also Dial and DialWithConfig configurations. 

What could be the reason ?

**Comment by zygis** [comment]
Try this:
`mongodb://user:pass@prefix1.mongodb.net:27017,prefix2.mongodb.net:27017,prefix3.mongodb.net:27017/admin`

**Comment by logicaleak** [comment]
Nope, does not solve it at all. Let me tell you how I solved it. I have started reading the source code, and noticed acquireSocket simply cannot acquire the socket, and the ping fails. Hence, as everything about the addresses looked just fine, I have suspected it could be because of the way the network connection is made. With the help of a random blog post about activating ssl for mgo, I have come up with this : 

```
import "crypto/tls"

dialInfo, err := mgo.ParseURL("mongodb://user:pass@prefix.mongodb.net:27017,prefix.mongodb.net:27017,prefix.mongodb.net:27017/admin?replicaSet=Cluster0-shard-0&authSource=admin")
	dialInfo.DialServer = func(addr *mgo.ServerAddr) (net.Conn, error) {
		conn, err := tls.Dial("tcp", addr.String(), tlsConfig)
		return conn, err
	}
	session, err := mgo.DialWithInfo(dialInfo)
	return session, errors.Wrap(err, "Could not dial mongo")
```

We could even activate ssl if we want to this way. Which is great.

**Comment by gyurobenjamin** [comment]
Does anyone solved this issue??? I'm getting same error now, and I have no idea what I can do.

**Comment by lloistborn** [comment]
you should close this issue since it worked for me too @logicaleak
